portfolio/src/main/webapp/script.js

```diff
@@ -27,64 +28,55 @@ function generateImageUrl() {
  * functionality but with a new, random image. 
  */
 function newImageButton() {
-
     const pennyButton = document.getElementById('penny-button');
-    const currentImgUrl = pennyButton.getAttribute("src");
+    const currentImgUrl = pennyButton.getAttribute('src');
     let newImgUrl = currentImgUrl;
 
-    //ensure new image is not the same as old one 
+    /* ensure new image is not the same as old one */ 
     while (newImgUrl === currentImgUrl) {
         newImgUrl = generateImageUrl();
     }
 
     const pennyContainer = document.getElementById('penny-container');
-    let newHTML = `<input id="penny-button" type="image" src="${newImgUrl}"\
-     onClick="newImageButton()"/>`;
+    let newHTML = `<input id="penny-button" type="image" src="${newImgUrl}"` +
+        ` onClick="newImageButton()"/>`;
     pennyContainer.innerHTML = newHTML;
 }
 
 /**
-  * Automatically fill gallery page with all available images
+  * Automatically fills gallery page with all available images
   */
 function fillGallery() {
     const imgsContainer = document.getElementById('penny-gallery-images');
-
     let htmlToAdd = '';
     for (let i = 1; i <= NUM_AVAILABLE_IMAGES; i++) {
-
-        htmlToAdd += `<input type="image" src="images/Penny-${i}.JPG"\ 
-            onClick="enlargeThisImage(${i})"/> `
+        htmlToAdd += `<input type="image" src="images/Penny-${i}.JPG"` + 
+            ` onClick="enlargeThisImage(${i})"/> `
     }
-
     imgsContainer.innerHTML = htmlToAdd;
 }
 
 /**
-  * Display `thisImage` enlarged above a grid of the other available images. 
+  * Displays `thisImage` enlarged above a grid of the other available images. 
   * `thisImage` becomes a link to the image, while all of the other images
   * are buttons that call this function. 
-  * @param int the image to be enlarged
+  * @param {int} thisImage The image to be enlarged
   */
 function enlargeThisImage(thisImage) {
     const imgsContainer = document.getElementById('penny-gallery-images');
-    const enlargedContainer = document.getElementById('enlarged-image');
-    const smallContainer = document.getElementById('small-images');
-
     imgsContainer.innerHTML = ''; 
 
-    enlargedContainer.innerHTML = `<a href="images/Penny-${thisImage}.JPG">\
-            <img src="images/Penny-${thisImage}.JPG"/></a> `
-
+    const enlargedContainer = document.getElementById('enlarged-image');
+    enlargedContainer.innerHTML = `<a href="images/Penny-${thisImage}.JPG">` +
+            `<img src="images/Penny-${thisImage}.JPG"/></a> `
 
+    const smallContainer = document.getElementById('small-images');
     let htmlToAdd = '';
     for (let i = 1; i <= NUM_AVAILABLE_IMAGES; i++) {
```

Comment:
one possible improvement here could be the duplicated logic between fillGallery and this method. 

See https://gist.github.com/davidgaskins/256860991fefaab7bdb814e441248cce for an example of what this could look like.

Reply:
I'm going to try doing this, I'll let you know when it's pushed

Reply:
wow, now that it's written it's so much more elegant! Good idea. I took out both for loops, but left in the other two instances of HTML generation (my rationale for doing this is that they were different enough that I would either have to make several different functions to handle their generation or make a single function with many parameters and possibly if statements. I thought it might clutter up the code and make it more difficult to read). If you want me to write functions for those two instances as well, I totally can!

Reply:
Oh... I didn't see before how different they are. 

Hmm... I wouldn't suggest putting them all in one function, but I would still suggest making all a function for each piece of html you generate. 

// Generators. 
enlargeImage()
createImage()
[i think there might be another?]

The idea here is to separate all the logic so that a reader can observe solely from the interface what is going on. I do want to point out that this is all *improvements* and the project output that you have is very good already.

While I'm thinking about improvements, I have something else for you to consider. It looks like you might be trying to regenerate the click handlers for certain items (not 100% sure). Instead, I would suggest names like "onEnlargeImage", "on...". that way there's a way to tell which methods are a result of user actions and which are simply methods that are there for modularization purposes.

Reply:
I made the edits!